File: backend/src/models/Content.js
const mongoose = require('mongoose');

const contentSchema = new mongoose.Schema(
  {
    title: {
      type: String,
      required: [true, 'El título del contenido es obligatorio'],
      trim: true,
    },
    description: {
      type: String,
      required: [true, 'La descripción es obligatoria'],
    },
    type: {
      type: String,
      enum: ['movie', 'series', 'documentary'], // Tipos de contenido
      required: [true, 'El tipo de contenido es obligatorio'],
    },
    genres: [
      {
        type: String,
        trim: true, // Ej: ['Acción', 'Drama', 'Sci-Fi']
      },
    ],
    videoUrl: {
      type: String,
      required: [true, 'La URL del video es obligatoria'],
    },
    thumbnailUrl: {
      type: String,
      default: '', // Imagen de miniatura o póster
    },
    duration: {
      type: Number,
      default: 0, // Duración en minutos
    },
    releaseYear: {
      type: Number,
    },
    isPremium: {
      type: Boolean,
      default: false, // Si es 'true', solo usuarios con subscriptionPlan 'premium' o 'pro' pueden verlo
    },
    isActive: {
      type: Boolean,
      default: true, // Permite ocultar el contenido sin borrarlo de la base de datos
    },
    views: {
      type: Number,
      default: 0, // Contador de reproducciones
    }
  },
  {
    timestamps: true, // Añade createdAt y updatedAt
  }
);

module.exports = mongoose.model('Content', contentSchema);

File: frontend/santv.app/lib/models/content.dart
class ContentItem {
  final String id;
  final String title;
  final String description;
  final String type;
  final List<String> genres;
  final String videoUrl;
  final String thumbnailUrl;
  final int duration;
  final int? releaseYear;
  final bool isPremium;
  final int views;

  const ContentItem({
    required this.id,
    required this.title,
    required this.description,
    required this.type,
    required this.genres,
    required this.videoUrl,
    required this.thumbnailUrl,
    required this.duration,
    this.releaseYear,
    required this.isPremium,
    required this.views,
  });

  bool get isYoutube =>
      videoUrl.contains('youtube.com') || videoUrl.contains('youtu.be');

  factory ContentItem.fromJson(Map<String, dynamic> json) {
    return ContentItem(
      id: json['_id']?.toString() ?? '',
      title: json['title']?.toString() ?? '',
      description: json['description']?.toString() ?? '',
      type: json['type']?.toString() ?? '',
      genres: (json['genres'] as List<dynamic>? ?? [])
          .map((g) => g.toString())
          .toList(),
      videoUrl: json['videoUrl']?.toString() ?? '',
      thumbnailUrl: json['thumbnailUrl']?.toString() ?? '',
      duration: (json['duration'] ?? 0) as int,
      releaseYear: json['releaseYear'] as int?,
      isPremium: json['isPremium'] as bool? ?? false,
      views: (json['views'] ?? 0) as int,
    );
  }
}
